packages/api/lib/executions.js

```diff
@@ -0,0 +1,55 @@
+const {
+  getKnexClient,
+  executionArnsFromGranuleIdsAndWorkflowNames,
+} = require('@cumulus/db');
+
+const { RecordDoesNotExist } = require('@cumulus/errors');
+
+/**
+ *  Finds and returns alternative executionArn related to the input granuleId.
+ *  Used to override the default (latest) executionArn when reingesting granules.
+ *  The decision tree is simple.
+ *  1. If a user inputs an executionArn we return it.
+ *  2. If not and no workflowName is specified, we return undefined so that the
+ *  granule's original execution is retained during reingest.
+ *  3. if not and a workflowName is input, we search the database for all
+ *  executions that match the granuleId and workflowName and return the most
+ *  recent.
+ *
+ * @param {Object} params - function parameters
+ * @param {string} params.granuleId - granuleId
+ * @param {string|undefined} [params.executionArn] - execution arn to use for reingest
+ * @param {string|undefined} [params.workflowName] - workflow name to use for reingest
+ * @param {function|undefined} [params.dbFunction] - database function for
+ *     testing. Defaults to executionArnsFromGranuleIdsAndWorkflowNames.
+ * @returns {Promise<string>|Promise<undefined>} - executionArn used in a
+ *             granule reingest call to determine correct workflow to run or
+ *             undefined.
+ */
+const chooseTargetExecution = async ({
+  granuleId,
+  executionArn = undefined,
+  workflowName = undefined,
+  dbFunction = executionArnsFromGranuleIdsAndWorkflowNames,
+}) => {
+  // if a user specified an executionArn, use that always
+  if (executionArn !== undefined) return executionArn;
+  // if a user didn't specify a workflow, return undefined explicitly
+  if (workflowName === undefined) return undefined;
```

Comment:
This goes with the thinking face above.  It might simplify it.  If I just return granule.execution I would have the same behavior, But is it more confusing knowing that chooseTargetExecution will just return the input execution... 🤔

Reply:
I also don't have the original granule execution in here.  I guess I kinda like the explicitness that we have overridden the default execution with a new one.  But I'm not married to that.